src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/CmdLineClient.kt

```diff
@@ -101,4 +107,5 @@ fun main(args: Array<String>) {
     val publishData = getPublishData(zippedData, parsedArgs.rootFileName, taskId)
 
     Publisher(topic).publish(publishData, onFailureCallback)
+    SubscribeManager(parsedArgs.resultTopic, ResultReceiver()).subscribe()
```

Comment:
Just extract code which attaches subscriber and manipulates with shutdown hook into a helper function and remove those class casts from SubscribeManager and usage of SubscribeManager from here.

Reply:
Done